tests/py/dynamo/models/test_engine_cache.py

```diff
@@ -41,17 +44,141 @@ def save(
+                max_shape=(200, 3, 224, 224),
+            ),
+        )
+        settings2 = CompilationSettings(
+            make_refittable=True, cache_built_engines=True, reuse_cached_engines=True
+        )
+        hash2 = BaseEngineCache.get_hash(exp_program2.module(), input_specs2, settings2)
+
+        self.assertEqual(hash1, hash2)
+
+    def test_input_shape_change_is_not_equal(self):
+        pyt_model = models.resnet18(pretrained=True).eval().to("cuda")
+        example_inputs = (torch.randn((100, 3, 224, 224)).to("cuda"),)
+        batch = torch.export.Dim("batch", min=1, max=200)
+
+        exp_program1 = torch.export.export(
+            pyt_model, args=example_inputs, dynamic_shapes={"x": {0: batch}}
+        )
+        input_specs1 = (
+            torch_trt.Input(
+                min_shape=(1, 3, 224, 224),
+                opt_shape=(100, 3, 224, 224),
+                max_shape=(200, 3, 224, 224),
+            ),
+        )
+        settings1 = CompilationSettings(
+            make_refittable=True, cache_built_engines=True, reuse_cached_engines=True
+        )
+        hash1 = BaseEngineCache.get_hash(exp_program1.module(), input_specs1, settings1)
+
+        exp_program2 = torch.export.export(
+            pyt_model, args=example_inputs, dynamic_shapes={"x": {0: batch}}
+        )
+        input_specs2 = (
+            torch_trt.Input(
+                min_shape=(1, 3, 300, 300),
+                opt_shape=(100, 3, 300, 300),
+                max_shape=(200, 3, 300, 300),
+            ),
+        )
+        settings2 = CompilationSettings(
+            make_refittable=True, cache_built_engines=True, reuse_cached_engines=True
+        )
+        hash2 = BaseEngineCache.get_hash(exp_program2.module(), input_specs2, settings2)
+
+        self.assertNotEqual(hash1, hash2)
+
+    def test_engine_settings_is_not_equal(self):
+        pyt_model = models.resnet18(pretrained=True).eval().to("cuda")
+        example_inputs = (torch.randn((100, 3, 224, 224)).to("cuda"),)
+        batch = torch.export.Dim("batch", min=1, max=200)
+
+        exp_program1 = torch.export.export(
+            pyt_model, args=example_inputs, dynamic_shapes={"x": {0: batch}}
+        )
+        input_specs1 = (
+            torch_trt.Input(
+                min_shape=(1, 3, 224, 224),
+                opt_shape=(100, 3, 224, 224),
+                max_shape=(200, 3, 224, 224),
+            ),
+        )
+        settings1 = CompilationSettings(
+            make_refittable=True,
+            cache_built_engines=True,
+            reuse_cached_engines=True,
+            enabled_precisions={torch.float32},
+        )
+        hash1 = BaseEngineCache.get_hash(exp_program1.module(), input_specs1, settings1)
+
+        exp_program2 = torch.export.export(
+            pyt_model, args=example_inputs, dynamic_shapes={"x": {0: batch}}
+        )
+        input_specs2 = (
+            torch_trt.Input(
+                min_shape=(1, 3, 300, 300),
```

Comment:
seems this function is testing `CompilationSettings` only, so should keep input_specs same?